### DISCUSSION TITLE
Homepage widget integration

### DISCUSSION THREAD

**Discussion by Aloe-recite** [WGDashboard/WGDashboard#451] [Ideas]
Hi,
do you have any plan to add [Homepage](https://gethomepage.dev) ([Github](https://github.com/gethomepage/homepage)) widget integration? That would be awesome. I'd love to see stats like the wg-easy widget, very informative. I'm not sure if this is something you need to implement on both sides (I suppose so?) but maybe you could make a contribution to their code for this? I asked in their Github discussions, got a couple of upvotes but no comment / reply unfortunately. Homepage is very widespread so it could be worth it :)
Thanks and best, keep up the good work!

**Comment by Aloe-recite** [comment]
Auto-commenting: seems like Homepage as a custom API integration: [](https://gethomepage.dev/widgets/services/customapi/)
Maybe WgDashboard could just leverage that, I'm not sure how to do that but once figured out might simply be added in the. documentation so that users can profit easily.

**Comment by donaldzou** [comment]
Hi @Aloe-recite , yes I think some other users mentioned this too! I'm planning to implement this into the next release but not the upcoming as is already feature freeze :)

**Reply by Aloe-recite** [reply]
great news! Btw, I spent more than 2h yesterday trying to make it work by using WGDashboard API and the customapi of Homepage, it's my first time trying it out and of course I couldn't manage to achieve anything, I was thinking about implementing an intermediate endpoint but it's very likely just an error on my part. Maybe it's not even necessary to modify WGD that much and it's just a matter of understanding and documenting how to configure it with Homepage. Thanks for your efforts!
BTW, let me hijack my own thread: would it be possible to have and easier solution to be sure that WGD boots with the configurations ON or OFF as they were left before? WHile fiddling with it I noticed that it boots with the config OFF, I saw that it is documented how to overcome this but still it looks easier and more logical to me to have either a user option or a default behavior of booting in the same state it was before. Hope that makes sense. Thanks anyway!

**Reply by donaldzou** [reply]
Ah yes sometimes it will take quite a while to understand other program's documentation lol. Yes in the upcoming `v4.1` will allow you to set which configuration will automatically enable after starting WGDashboard!
